worldedit-core/src/main/java/com/sk89q/worldedit/function/entity/ExtentEntityCopy.java

```diff
@@ -104,6 +106,17 @@ public boolean apply(Entity entity) throws WorldEditException {
             // Some entities store their position data in NBT
             state = transformNbtData(state);
 
+            // Paintings are very special
+            if (state.getType() == EntityTypes.PAINTING) {
```

Comment:
Why do you need this? It doesn't make a ton of sense to me, since the Tile position was just updated to where the location should be. If this is inconsistent, we should probably change how `newLocation` is calculated, not read it out of NBT data we just wrote.

Reply:
We need this, because getLocation for (vanilla) paintings is inconsistent. The calculation is perfectly fine, just Mojang adds some strange offset. When setting the location of a painting, it updates TileXYZ to the values set and then updates the location to something different.

This is a special thing that just exists for paintings.

Reply:
Alright, then:

>we should probably change how newLocation is calculated, not read it out of NBT data we just wrote.

Reply:
I have no idea how I should do that. Only the nbt data contains the correct values.

Reply:
Just re-use the existing code that makes the NBT values in `transformNbtData`? If that's not correct, then this won't be either. We haven't written back to MC yet, so it can't have made changes.

Reply:
The code uses the old NBT values to transform them. I could use the old NBT values and transform them again the same way as in `transformNbtData`, but would that be any better than this change?

Reply:
Ok, I changed that. Is it ok now?